Dockerfile

```diff
@@ -20,32 +20,28 @@ RUN apt-get update -y && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/*
 
+
+# Install node from upstream, ubuntu packages are too old
+RUN curl -sL https://deb.nodesource.com/setup_18.x | bash
```

Comment:
May I ask for confirmation that this will install nodejs v18?

It looks obvious to be true, but I want to be convinced that we can remove the later `RUN` commands for added nodejs v18 in the previous Dockerfile (left). 

```Dockerfile
RUN npm cache clean -f
RUN npm install -g n
RUN n 18.17.1
```

Note that nodejs is well outside of my experience, such the most obvious answer to this question could be:
"Yes, of course. Why are you asking?"

Reply:
All good, better double check. 

Nodesource is a nodejs binary distribution explicitly referenced in the official nodejs install documentation https://nodejs.org/en/download/package-manager#debian-and-ubuntu-based-linux-distributions

They provide `setup_VERSION` scripts for major nodejs versions. So yes, this installs the most current nodejs 18 directly.

Your approach also worked, what you basically did was:
 - taking the nodejs 20 already installed
 - using its package manager `npm` to install `n` - a nodejs version manager https://www.npmjs.com/package/n
 - using `n` to install nodejs 18 as an additional node version